checkov/sca_image/runner.py

```diff
@@ -196,6 +197,16 @@ def get_image_report(self, dockerfile_path: str, image: Image, runner_filter: Ru
             result = cached_results.get('results', [{}])[0]
             vulnerabilities = result.get("vulnerabilities") or []
             image_id = self.extract_image_short_id(result)
+            image_packages = result.get('packages', [])
+            image_package_types = {}
+            for package in image_packages:
+                image_package_types[f'{package["name"]}@{package["version"]}'] = package['type']
```

Comment:
```suggestion
            image_package_types = {
                f'{package["name"]}@{package["version"]}': package['type']
                for package in image_packages
            }
```
dict comprehension 🚀